### PR TITLE
chore(RHTAPWATCH-283): instructions for deploying Splunk instance

### PR DESCRIPTION
**Pull request by Omeramsc**
Adds a guide on how to deploy a containerized Splunk instance. 
This will be a part of a future testing platform for Splunk.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
   the SSO user ID to be sent to Segment
 - `status.compliantUsername` - Contains the username used in the cluster audit
   log.
+
+## Setting up a containerized Splunk instance
+- Either podman or docker can be user
```
Comment (ifireball) [inline README.md:119]:
```suggestion
- Either podman or docker can be used
```

File: README.md [hunk 2]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
...
+- running the following command will deploy a Splunk enterprise container:  
+- `podman pull docker.io/splunk/splunk:latest`
```
Comment (ifireball) [inline README.md:121]: No need to pre-pull, podman run can pull already.

File: README.md [hunk 3]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
...
+  `podman run -p 8000:8000 -e "SPLUNK_PASSWORD=AcceptablePassword1" -e "SPLUNK_START_ARGS=--accept-license" -it --name splunk_container docker.io/splunk/splunk:latest`
+- More info about the specific inputs of the command can be found [here|https://docs.splunk.com/Documentation/Splunk/9.0.4/Installation/DeployandrunSplunkEnterpriseinsideDockercontainers]
```
Comment (ifireball) [inline README.md:123]: Use the [foo][bar] link syntax do make the file readable in text form. Also this syntax is not markdown at all, it looks like the ASCIIDOC syntax used by Jira.

File: README.md [hunk 4]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
...
+
+
+## Setting up a containerized Splunk instance
```
Comment (ifireball) [inline README.md:126]: duplicate header

File: README.md [hunk 5]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
...
+1. Pull the latest Splunk Enterprise container image:
+   ```
+   podman pull docker.io/splunk/splunk:latest
+   ```
+
+2. Run the container with the desired configuration:
+   ```
+   podman run -p 8000:8000 -e "SPLUNK_PASSWORD=YourPassword" -e "SPLUNK_START_ARGS=--accept-license" -it --name splunk_container docker.io/splunk/splunk:latest
+   ```
+   This command maps port 8000 from the container to the host machine, sets the Splunk password to meet the required complexity, and accepts the Splunk license.
```
Comment (ifireball) [inline README.md:139]: What is the required complexity? Don't make me have to follow a link to know that!

File: README.md [hunk 6]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
...
+
+   For more information about the specific command inputs and options, refer to the [Splunk documentation](https://docs.splunk.com/Documentation/Splunk/9.0.4/Installation/DeployandrunSplunkEnterpriseinsideDockercontainers).
+
```
Comment (ifireball) [inline README.md:142]: No need to double space

File: README.md [hunk 7]
```diff
@@ -114,3 +114,45 @@ The interesting fields for this use case are:
...
+
+3. Once the container is running, you can log in to the Splunk instance using the username `admin` and the password you set up.
+
+
+4. To access the Splunk UI, open a web browser on the host and navigate to `localhost:8000`.
+
+
+* If you prefer to use the Splunk REST API, follow these steps:
+   - Get inside the running container using the command:
```
Comment (ifireball) [inline README.md:151]: Why do I have to do this? Pleas also expose port 8089 so we can do the curl calls directly from the host.

Also please add a link to the API docs somewhere.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Omeramsc) [reply README.md]: It's a mystery how I didn't notice all the leftover duplicated text after re-reading it 3  times... fixed.

File: README.md (hunk 2 above)
Comment (Omeramsc) [reply README.md]: removed

File: README.md (hunk 3 above)
Comment (Omeramsc) [reply README.md]: leftover, removed

File: README.md (hunk 4 above)
Comment (Omeramsc) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (Omeramsc) [reply README.md]: changed

File: README.md (hunk 6 above)
Comment (Omeramsc) [reply README.md]: removed

File: README.md (hunk 7 above)
Comment (Omeramsc) [reply README.md]: done

File: README.md [hunk 8]
```diff
@@ -114,3 +114,23 @@ The interesting fields for this use case are:
...
+
+To set up a containerized Splunk instance, you can use either podman or docker. Follow these steps:
+
+1. Run the container with the desired configuration:
+   ```
+   podman run -p 8000:8000 -p 8089:8089 -e "SPLUNK_PASSWORD=YourPassword" -e "SPLUNK_START_ARGS=--accept-license" -it --name splunk_container docker.io/splunk/splunk:latest
+   ```
+   This command maps port 8000 from the container to the host machine, set a password that is a minimum of 8 characters long (the minimum requirement), and accepts the Splunk license.
+
+   For more information about the specific command inputs and options, refer to the [Splunk documentation](https://docs.splunk.com/Documentation/Splunk/9.0.4/Installation/DeployandrunSplunkEnterpriseinsideDockercontainers).
```
Comment (ifireball) [inline README.md:128]: Please use the reference-and-footnote link syntax to improve text-mode readability. 

```suggestion
   For more information about the specific command inputs and options, refer to the [Splunk documentation][CS1].

 ...
         
[CS1]: https://docs.splunk.com/Documentation/Splunk/9.0.4/Installation/DeployandrunSplunkEnterpriseinsideDockercontainers
```

Also please break your lines at the 80 chars marker. `vim` can do this with `gqgq` while other editory probably support this as well.

File: README.md [hunk 9]
```diff
@@ -114,3 +114,23 @@ The interesting fields for this use case are:
...
+
+## Setting up a containerized Splunk instance
+
+To set up a containerized Splunk instance, you can use either podman or docker. Follow these steps:
+
+1. Run the container with the desired configuration:
+   ```
+   podman run -p 8000:8000 -p 8089:8089 -e "SPLUNK_PASSWORD=YourPassword" -e "SPLUNK_START_ARGS=--accept-license" -it --name splunk_container docker.io/splunk/splunk:latest
+   ```
+   This command maps port 8000 from the container to the host machine, set a password that is a minimum of 8 characters long (the minimum requirement), and accepts the Splunk license.
```
Comment (ifireball) [inline README.md:126]: There is not enough indentation here to make this line be a sup-paragraph of the bullet paragraph - you need at least 4.

File: README.md [hunk 10]
```diff
@@ -114,3 +114,23 @@ The interesting fields for this use case are:
...
+
+   For more information about the specific command inputs and options, refer to the [Splunk documentation](https://docs.splunk.com/Documentation/Splunk/9.0.4/Installation/DeployandrunSplunkEnterpriseinsideDockercontainers).
+2. Once the container is running, you can log in to the Splunk instance using the username `admin` and the password you set up.
```
Comment (ifireball) [inline README.md:129]: The text alignment here seems to be off.

**Review by ifireball — COMMENTED** [review]
Some stylistic comments

File: README.md (hunk 10 above)
Comment (Omeramsc) [reply README.md:129]: Not sure what you refer to?

File: README.md (hunk 8 above)
Comment (Omeramsc) [reply README.md:128]: huh, didn't know that's a thing. cool. changed

File: README.md (hunk 9 above)
Comment (Omeramsc) [reply README.md]: fixed (I think?)

File: README.md [hunk 11]
```diff
@@ -159,3 +159,33 @@ then 4 times in a row and issue an appropriate alert.
...
+    This command maps port 8000 from the container to the host machine, set a
+    password that is a minimum of 8 characters long (the minimum requirement),
+    and accepts the Splunk license.
+    For more information about the specific command inputs and options, refer
+    to the [Splunk documentation][CS1].
+2. Once the container is running, you can log in to the Splunk instance using
+   the username `admin` and the password you set up.
+3. To access the Splunk UI, open a web browser on the host and navigate to
+   `localhost:8000`.
+* If you prefer using the Splunk [REST API][CS2],
```
Comment (ifireball) [inline README.md:181]: is this bullet supposed to be nested under `3.` ?

File: README.md [hunk 12]
```diff
@@ -159,3 +159,33 @@ then 4 times in a row and issue an appropriate alert.
 [ES5]: https://docs.splunk.com/Documentation/SplunkCloud/9.0.2303/SearchReference/Transaction
 [ES6]: https://docs.splunk.com/Documentation/Splunk/9.0.4/SearchReference/Eval
 [ES7]: https://docs.splunk.com/Documentation/Splunk/9.0.4/SearchReference/JSONFunctions#json_object.28.26lt.3Bmembers.26gt.3B.29
+
+## Setting up a containerized Splunk instance
```
Comment (ifireball) [inline README.md:163]: To make things clearer, perhaps we should make a "running a test environment" section and then put this as a sub section under that.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (Omeramsc) [reply README.md:181]: I wrote it as either access the UI or use the REST API, as they *kind of* perform the same function, but separating the bullets also make sense.

File: README.md (hunk 12 above)
Comment (Omeramsc) [reply README.md]: fixed

**Pull request merged**